mempool/mempool.go

```diff
@@ -310,14 +311,25 @@ func (mp *TxPool) removeOrphan(txHash *chainhash.Hash) {
 
 	// Remove the reference from the previous orphan index.
 	for _, txIn := range tx.MsgTx().TxIn {
-		originTxHash := txIn.PreviousOutPoint.Hash
-		if orphans, exists := mp.orphansByPrev[originTxHash]; exists {
-			delete(orphans, *tx.Hash())
+		orphans, exists := mp.orphansByPrev[txIn.PreviousOutPoint]
+		if exists {
+			delete(orphans, *txHash)
 
 			// Remove the map entry altogether if there are no
 			// longer any orphans which depend on it.
 			if len(orphans) == 0 {
-				delete(mp.orphansByPrev, originTxHash)
+				delete(mp.orphansByPrev, txIn.PreviousOutPoint)
+			}
+		}
+	}
+
+	// Remove any orphans that redeem outputs from this one if requested.
+	if removeRedeemers {
+		prevOut := wire.OutPoint{Hash: *txHash}
```

Comment:
This section is not correct for Decred since it ignores the `Tree`.  It will miss stake transactions which have a `Tree` of `wire.TxTreeStake`.

As discussed offline, I would suggest creating a `struct treelessOutPoint` which consists of only the hash and use that structure in mempool instead, then convert the `wire.OutPoints` accordingly when working with them.